Animal Competition/Java/Game.java

```diff
@@ -0,0 +1,327 @@
+            if (animal1Points>animal2Points){
+              this.show("Animal 1 Wins!");
+            } else if (animal1Points<animal2Points){
+              this.show("Animal 2 Wins!");
+            }
+
+
+              
+          else if (desertSituationNumber == 1){
+            
+            int compareAttacking = this.attackingSuperiority(animal1AttackingAbility, animal2AttackingAbility);
+            
+            if (compareAttacking > 0)
+              animal1Points++;
+            else if (compareAttacking < 0)
+              animal2Points++;
+            else // they are equal
+            {
+              animal1Points++;
+              animal2Points++;
+            }
+            
+            // compares running ability for animals
+            int compareRunning1 = this.runningSuperiority(animal1RunningAbility, animal2RunningAbility);
+            if (compareRunning1 > 0)
+              animal1Points++;
+            else if (compareRunning1 < 0)
+              animal2Points++;
+            else{
+              animal1Points++;
+              animal2Points++;
+            }
+            
+            // compares burrowing ability for animals
+            int compareFlying1 = this.burrowingSuperiority(animal1BurrowingAbility, animal2BurrowingAbility);
+            if (compareFlying1 > 0)
+              animal1Points++;
+            else if (compareFlying1 < 0)
+              animal2Points++;
+            else{
+              animal1Points++;
+              animal2Points++;
+            }
+            
+            // tallies total points and determines a winner
+            if (animal1Points>animal2Points){
+              this.show("Animal 1 Wins!");
+            } else if (animal1Points<animal2Points){
+              this.show("Animal 2 Wins!");
+            }
+          }
+            
+          }
+          
+          
+          
+          }
+          else if (environmentNumber == 0){
+
+          String forestSituation0 = "The animals have to find food in the forest!";
+
+          String forestSituation1 = "The animals have to avoid a predator in the forest!";
+
+          String forestSituation2 = "The animals have to surive a flash flood in the forest!";
+            
+          var forestSituations = new ArrayList<String>();
+          forestSituations.add(forestSituation0); 
+          forestSituations.add(forestSituation1);
+          forestSituations.add(forestSituation2);
+          int randomNum2 = (int)(Math.random() * 3);
+
+          this.show(forestSituations.get(randomNum2));
+
+          
+        }
+
+          
+            
+        
+
+      
+      
+        
+      
+
+      
+    }
+    }
+
+
+
+
+void show(String text){
+  System.out.println(text);
+}
+
+// Determines if the first trait is superior to the second trait:
+// 1 for yes, -1 for no, 0 for equality
+// climbing ability comparison
+int climbingSuperiority(int animal1ClimbingAbility, int animal2ClimbingAbility){
```

Comment:
All these `xxxSuperiority` functions are identical except for the names of the parameter.  You just need one and passing in different parameters.